### DISCUSSION TITLE
Support for 2nd EON frequency in AF/EON view in case of multiple tuning frequency

### DISCUSSION THREAD

**Discussion by andimik** [PE5PVB/TEF6686_ESP32#393] [Ideas]
replaces  #387 

I've realized that the EON Frequency in AF/EON view might not be correct as the tuned frequency is used more than once.

In Advanced View there is already support for multiple EON frequency possibilities.

![IMG_20231103_203153](https://github.com/PE5PVB/TEF6686_ESP32/assets/24510556/c46362bd-ffd5-4668-8e52-bc92950d132f)

In my example I have tuned to 94.7 (RADIO-K is described as OE2 K in the below screenshots) which is used twice here. So the EON frequency for OE 1 can be either 87.7 or 89.7. 

I've checked the official frequency table

![image](https://github.com/PE5PVB/TEF6686_ESP32/assets/24510556/33411454-d6e6-44f9-880b-db7ce3e2c193)

![image](https://github.com/PE5PVB/TEF6686_ESP32/assets/24510556/e7415a62-2c88-43a1-9977-9583a72a299a)

In RDS, both are mentioned:

```
{"pi":"0xA502","group":"14A","other_network":{"pi":"0xA201","kilohertz":89700,"tp":false},"prog_type":"No PTY","tp":true}
{"pi":"0xA502","group":"14A","other_network":{"pi":"0xA201","kilohertz":87700,"tp":false},"prog_type":"No PTY","tp":true}
```

So I propose that both possible frequencies are either shown alternating or write a second line with the frequency and in front the word `or`

```
A201   OE 1       87.7
           or     89.7
```

**Comment by lxsxl** [comment]
In addition I suggest to merge this thread with this one https://github.com/PE5PVB/TEF6686_ESP32/discussions/337.

**Comment by PE5PVB** [comment]
I have a stream from WDR2, SWR3 and Bayern3. Does any of them contain a double EON refernce so I can test?

**Comment by lxsxl** [comment]
There was an mpx-file from 91,8 / WDR 2 / D392. As far as I remember it contains three EON-frequencies per referenced network. Do you still have it or shall I upload it again?

**Comment by PE5PVB** [comment]
I rebuild the AF/EON part.
First page is now only AF (up to 50 entries)
Second page: EON page 1 with PI, PS/ 3x mapped frequency and TA, TP indicator.
Thirth page: EON page 2 when more than 10 entries.

**Comment by lxsxl** [comment]
Also this has become really great! Three EON-frequencies work very fine, also TP and TA is nice. If you move PS and EON-AF close to PI there might be just enough space for PTY as decimal or hex code. Would be the last EON finishing :-)

**Comment by lxsxl** [comment]
And now even with sorted EON networks ☺☺☺

**Comment by lxsxl** [comment]
May I ask again even though you closed this thread: If you move PS and EON-AF close to PI there might be just enough space for PTY as decimal or hex code. Would be the last EON finishing :-)

**Comment by PE5PVB** [comment]
Yes, I can add decimal value for PTY in EON.

**Comment by lxsxl** [comment]
That's perfect, thanks!

**Comment by andimik** [comment]
What about the 2nd column frequency? It seems this is the wrong color.

![_20231111_205955](https://github.com/PE5PVB/TEF6686_ESP32/assets/24510556/b9c82d60-9200-47eb-bb5a-de340f5323d8)

**Comment by lxsxl** [comment]
Isn't this intended? The colour of all columns are alternating. Or otherwise PS would be also the wrong colour.

**Comment by PE5PVB** [comment]
Yes, this is no error. It's intended to make it easier readable.

**Comment by andimik** [comment]
I don't agree.

**Reply by PE5PVB** [reply]
With what?

**Reply by andimik** [reply]
> It's intended to make it easier readable. 

I say no, because when there are two different colors we could think, the middle frequency is something special (like fillers). 

When there is only one color, all are the same.

**Reply by andimik** [reply]
Thanks for fixing that :-)

**Comment by lxsxl** [comment]
Just found a very strange thing (first freq. in last EON)
![IMG_8531](https://github.com/PE5PVB/TEF6686_ESP32/assets/140750201/5c1a87e8-d757-4945-9705-1a5d1f6f5347)

**Comment by lxsxl** [comment]
But apart from this bug EON is perfect now!

**Comment by andimik** [comment]
109,5 MHz? First I thought the number of frequencies is calculated wrong.

But 220 (which would be 109.5 MHz, if you continue in theory) is not assigned

![grafik](https://github.com/PE5PVB/TEF6686_ESP32/assets/24510556/4de33bb2-50f5-482b-805c-79b97b41c580)

> 204 | 107,9
205 | 108
206 | 108,1
207 | 108,2
208 | 108,3
209 | 108,4
210 | 108,5
211 | 108,6
212 | 108,7
213 | 108,8
214 | 108,9
215 | 109
216 | 109,1
217 | 109,2
218 | 109,3
219 | 109,4
220 | 109,5
221 | 109,6
222 | 109,7
223 | 109,8
224 | 109,9
225 | 110
226 | 110,1
227 | 110,2
228 | 110,3

**Comment by lxsxl** [comment]
Yes, this was also my first thought. But if a not assigned code is received it should be ignored.

**Comment by PE5PVB** [comment]
I've added some boundries at the frequency parsing.
